hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1698,7 +1701,19 @@ public boolean isEmpty() {
     }
 
     @Override
-    public void putAll(@Nonnull Map<? extends K, ? extends V> map) {
+    public void putAll(@Nonnull Map<? extends K, ? extends V> m) {
+        putAllInternal(m, null);
+    }
+
+    // used by Jet
+    public ICompletableFuture<Void> putAllAsync(@Nonnull Map<? extends K, ? extends V> m) {
```

Comment:
I would suggest we add `putAllAsync` in `IMap` interface, wdyt?

Reply:
I think we should, but my understanding is that we don't add methods unless some customer requests it. This was the case with async `executeOnKeys` that I added maybe a year ago and it's not public API.